// Copyright 2025 Google LLC.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Simple test service definition.
syntax = "proto3";

package test_cuda;

message TestConfig {}
service TestService {
  rpc TestCall(TestRequest) returns (TestResponse) {}
}
message InferenceCall {
  map<string, bytes> inputs = 1;
}
message InferenceCallResult {
  map<string, bytes> outputs = 1;
  double latency_seconds = 2;
}
message TestRequest {
  repeated InferenceCall inference_calls = 5;
}
message TestResponse {
  string msg = 1;
  repeated InferenceCallResult inference_call_results = 3;
}
